extends CharacterBody2D

@export var left_limit: float = -100   # Giá»›i háº¡n bÃªn trÃ¡i
@export var right_limit: float = 100   # Giá»›i háº¡n bÃªn pháº£i
@export var speed: float = 50          # Tá»‘c Ä‘á»™ bay

var direction := 1                     # 1 = sang pháº£i, -1 = sang trÃ¡i
var start_x := 0.0                       # LÆ°u vá»‹ trÃ­ ban Ä‘áº§u

@export var hp := 1

@onready var anim = $AnimatedSprite2D

func _ready():
	add_to_group("enemy")
	
	start_x = global_position.x
	anim.play("fly")  # animation bay

func _physics_process(_delta):
	# Di chuyá»ƒn, CÃ¡ch nÃ y Bá»Ž QUA collision hoÃ n toÃ n
	#global_position.x += speed * direction * delta
	#Muá»‘n va cháº¡m TileMap â†’ PHáº¢I dÃ¹ng:
	#velocity
	#move_and_slide() hoáº·c move_and_collide()
	velocity.x = speed * direction
	move_and_slide()

	# Äá»•i hÆ°á»›ng khi cháº¡m biÃªn
	if global_position.x > start_x + right_limit:
		direction = -1

	elif global_position.x < start_x + left_limit:
		direction = 1
		
	# Náº¿u Ä‘á»¥ng tÆ°á»ng thÃ¬ quay Ä‘áº§u
	if is_on_wall():
		direction *= -1

	# Quay Ä‘áº§u/Láº­t sprite theo hÆ°á»›ng bay
	anim.flip_h = direction > 0

func hit(damage: int):
	hp -= damage
	if hp <= 0:
		die()

func die():
	queue_free()   # ðŸ biáº¿n máº¥t
